### PR TITLE
Import Foundation in codegen

### PR DESCRIPTION
**Pull request by dfed**
Hi folks! The Swift code-gen today requires that an Objective-C umbrella header be present that has an `import <UIKit/UIKit.h>` to compile. You can see this for yourself by running code-gen on any request/response that has a `Date` as a value or parameter. `Date` comes from `Foundation`, but there is no import of `Foundation`. Instead, the generated code implicitly relies on the `UIKit` import from an umbrella header to pull in `Foundation`.

This PR explicitly imports `Foundation` in the generated code, which allows consumers of Swift Apollo code-gen to no longer need an umbrella header.

TODO:

- [x] Update CHANGELOG.md\* with your change (include reference to issue & this PR)
- [x] Make sure all of the significant new logic is covered by tests
- [x] Rebase your changes on master so that they can be merged easily
- [x] Make sure all tests and linter rules pass

\*Make sure changelog entries note which project(s) has been affected. See older entries for examples on what this looks like.

### CODE REVIEW

**Comment by trevor-scheer** [comment]
@designatednerd, would you mind taking a look at this when you get the chance? Thanks!

**Comment by fassko** [comment]
Would be great to get this forward. Thanks!

**Comment by designatednerd** [comment]
@dfed Do you have a bit of time to rebase this, or do you mind if I take over this PR?

**Comment by dfed** [comment]
Happy to have you take this PR over the finish line @designatednerd! I’m out on vacation this week.

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,7 +11,7 @@
 - `apollo-codegen-scala`
   - <First `apollo-codegen-scala` related entry goes here>
 - `apollo-codegen-swift`
-  - <First `apollo-codegen-swift` related entry goes here>
+  - Allow generated code to be compiled without a module umbrella header 1248](https://github.com/apollographql/apollo-tooling/pull/1248)
```
Comment (designatednerd) [inline CHANGELOG.md]: fwiw you don't need to put the link here

**Comment by designatednerd** [comment]
OK cool - ping me when it's ready for another look

**Comment by designatednerd** [comment]
🤦‍♀ Completely misread the earlier comment! Will take this over.

**Pull request merged**